### PR TITLE
React 18

### PR DESCRIPTION
**Pull request by ferothefox**
This PR upgrades React and ReactDOM to 18.2.0, which are the latest versions, and with it, the latest hooks and performance improvements. As far as I know, pyro is running the only production Pterodactyl panel running these versions of React. This will allow us to share our marketing website UI lib with our panel and begin optimizing pyrodactyl to be accessible and keyboard navigable. 

The vast majority of changes are simply updating imports across the entire project. The main changes are updating the swr API (which powers the live stats in the server Home page, and features like websockets for the backup page).

You should see considerably lower build times on this branch 🎉

### CODE REVIEW

**Comment by ChecksumDev** [comment]
Let me look over this, but already LGTM.

File: resources/scripts/components/dashboard/ServerRow.tsx
Comment (ChecksumDev) [inline resources/scripts/components/dashboard/ServerRow.tsx:105]: Customers might have a server description especially in cases like people with server networks, a lot of networks use ID systems separate from the panel that might be used here.

File: resources/scripts/components/elements/ProgressBar.tsx
Comment (ChecksumDev) [inline resources/scripts/components/elements/ProgressBar.tsx:1]: Bye bye loading bars!

File: resources/scripts/components/elements/SpinnerOverlay.tsx
Comment (ChecksumDev) [inline resources/scripts/components/elements/SpinnerOverlay.tsx:1]: Bye bye spinners!

File: resources/scripts/components/server/backups/CreateBackupButton.tsx
Comment (ChecksumDev) [inline resources/scripts/components/server/backups/CreateBackupButton.tsx:83]: Could this async lead to a race condition given this is a network operation? Not sure why this specifically pokes out to me.

**Review by ChecksumDev — COMMENTED** [review]
Just curious about a couple things, majority LGTM.

File: resources/scripts/components/server/backups/CreateBackupButton.tsx
Comment (ferothefox) [reply resources/scripts/components/server/backups/CreateBackupButton.tsx:83]: It shouldn't be a big deal since the req that creates a backup is only a couple bytes at its most. There should be no reason this fails and causes any major issues imo

File: resources/scripts/components/dashboard/ServerRow.tsx
Comment (ferothefox) [reply resources/scripts/components/dashboard/ServerRow.tsx:105]: You're right, I can add it back. Noted (put on linear eventually) and will revisit

File: resources/scripts/components/server/backups/CreateBackupButton.tsx
Comment (ChecksumDev) [reply resources/scripts/components/server/backups/CreateBackupButton.tsx:83]: Ahh, I thought it would be the size of the actual backup, all is well then ^^

**Review by ChecksumDev — APPROVED** [review]

**Comment by ChecksumDev** [comment]
Most of these changes are framework-specific. I'll proceed with the merge without a second review.

**Pull request merged**
